rpc/types_test.go

```diff
@@ -0,0 +1,76 @@
+package rpc_test
+
+import (
+	"testing"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/rpc"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBlockId(t *testing.T) {
+	tests := map[string]struct {
+		blockIdJson     string
+		expectedBlockId rpc.BlockId
+	}{
+		"latest": {
+			blockIdJson: "\"latest\"",
+			expectedBlockId: rpc.BlockId{
+				Latest: true,
+			},
+		},
+		"pending": {
+			blockIdJson: "\"pending\"",
+			expectedBlockId: rpc.BlockId{
+				Pending: true,
+			},
+		},
+		"number": {
+			blockIdJson: `{ "block_number" : 123123 }`,
+			expectedBlockId: rpc.BlockId{
+				Number: 123123,
+			},
+		},
+		"hash": {
+			blockIdJson: `{ "block_hash" : "0x123" }`,
+			expectedBlockId: rpc.BlockId{
+				Hash: new(felt.Felt).SetUint64(0x123),
+			},
+		},
+	}
+	t.Parallel()
```

Comment:
I believe this usage is incorrect you should move it to line 45. unless this was intentional.